### PR TITLE
Run the pytest testsuite in parallel

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - use `pytest-xdist` to run in parallel on 2 cores

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,15 +24,15 @@ env:
     - COVERALLS=false
     - NOSE_FLAGS="--processes=2 --process-timeout=400 --no-open-files --with-timer --timer-top-n 50"
     - NOSE_TEST_LIST="analysis auxiliary coordinates core formats topology utils"
-    - PYTEST_FLAGS="--disable-pytest-warnings"
+    - PYTEST_FLAGS="--disable-pytest-warnings -n 2"
```
Comment (kain88-de) [inline .travis.yml]: can you please use `--numprocesses` it's more explicit and I'm trying to get upstream ci-helpers to detect that to auto install pytest-xdist.

Comment (utkbansal) [inline .travis.yml]: Sure. I'll just let this build complete and then push.

**Comment by utkbansal** [comment]
@kain88-de Should be ready to merge. No drop in coverage.

File: .travis.yml (hunk 1 above)
Comment (utkbansal) [inline .travis.yml]: @kain88-de That doesn't work. I don't think there is anything like `--numprocesses`

**Comment by utkbansal** [comment]
This is weird, I think I saw no drop in coverage last night. Let me cross check.

**Comment by utkbansal** [comment]
I wonder how pytest shows percentage of coverage when it did not collect data.

**Comment by utkbansal** [comment]
@richardjgowers @jbarnoud @kain88-de @mnmelo I have a wild theory - the warnings might just be meaningless.

I ran pytest on the `lib` module with coverage, once serially and once in parallel - both times the coverage reported was the same. I assumed that the difference might be too small and it might be a round off error since the coverage was an integer.

So, I ran the `coordinates` module - because it is large in terms of the number of tests. The reasoning was that the difference in coverage would be larger and there should be a tangible difference even after rounding off to the nearest integer. I still got the same coverage in both cases - 41%

Commands I used - 

Parallel: `py.test --cov=MDAnalysis coordinates --cov-report=html -q --disable-pytest-warnings -n 2`
Serial: `py.test --cov=MDAnalysis coordinates --cov-report=html -q --disable-pytest-warnings`

Can someone run these commands locally (off the port coordinates PR) and confirm my findings?

**Comment by utkbansal** [comment]
So the build passed. The coverage is still at 88.837% Was there a drop? The button on the README file shows the same number.

File: .travis.yml [hunk 2]
```diff
@@ -24,15 +24,15 @@ env:
...
     - PYTEST_LIST="testsuite/MDAnalysisTests/lib testsuite/MDAnalysisTests/formats"
     - NOSE_COVERAGE_FILE="nose_coverage"
     - PYTEST_COVERAGE_FILE="pytest_coverage"
     - MAIN_CMD="pytest ${PYTEST_LIST} ${PYTEST_FLAGS}; python ./testsuite/MDAnalysisTests/mda_nosetests ${NOSE_TEST_LIST} ${NOSE_FLAGS}"
     - SETUP_CMD=""
     - BUILD_CMD="pip install -v package/ && pip install testsuite/"
-    - CONDA_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer pytest=3.1.2 pytest-cov=2.5.1"
-    - CONDA_ALL_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib netcdf4 scikit-learn scipy seaborn coveralls clustalw=2.1 pytest=3.1.2 pytest-cov=2.5.1"
+    - CONDA_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer pytest=3.1.2 pytest-cov=2.5.1 pytest-xdist=1.17.1"
```
Comment (kain88-de) [inline .travis.yml]: Could you please remove the version pinning. I'm trying to get everything updated to we can take use of ci-helpers magic to autoinstall things for us. You should be able to remove the pytest explicit dependency as well.

I'm hijacking this PR to clean up some installations of pytest.

**Comment by utkbansal** [comment]
@kain88-de @jbarnoud @richardjgowers I don't really like the way pytest displays its output (listing each file and the other stuff). I'd like it to be more like what nose does currently. Can I add the `-q` flag to minimize the noise?

File: .travis.yml [hunk 3]
```diff
@@ -24,15 +24,15 @@ env:
...
+    - PYTEST_FLAGS="--disable-pytest-warnings -n 2"
     - PYTEST_LIST="testsuite/MDAnalysisTests/lib testsuite/MDAnalysisTests/formats"
     - NOSE_COVERAGE_FILE="nose_coverage"
     - PYTEST_COVERAGE_FILE="pytest_coverage"
     - MAIN_CMD="pytest ${PYTEST_LIST} ${PYTEST_FLAGS}; python ./testsuite/MDAnalysisTests/mda_nosetests ${NOSE_TEST_LIST} ${NOSE_FLAGS}"
     - SETUP_CMD=""
     - BUILD_CMD="pip install -v package/ && pip install testsuite/"
-    - CONDA_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib scipy griddataformats pytest=3.1.2 pytest-cov=2.5.1"
-    - CONDA_ALL_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib netcdf4 scikit-learn scipy griddataformats seaborn coveralls clustalw=2.1 pytest=3.1.2 pytest-cov=2.5.1"
+    - CONDA_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib scipy griddataformats pytest-cov pytest-xdist"
```
Comment (kain88-de) [inline .travis.yml]: @utkbansal you should also be able to drop the `pytest-cov` package as an explicit dependency. See https://github.com/astropy/ci-helpers/pull/216

Comment (utkbansal) [inline .travis.yml]: Cool.

**Comment by richardjgowers** [comment]
@utkbansal rebase this and we're good to go

**Comment by utkbansal** [comment]
@richardjgowers The rebase for this depends on what we want to do about #1414 

It wasn't supposed to be merged yet. 

* Should I open a new PR for the `coordinates` module and fix coverage in that? Then I can drop `coordinates` from NOSE_TEST_LIST here

* Or if we will revert #1414 then, I have to drop `coordinates` from PYTEST_TEST_LIST.

**Comment by utkbansal** [comment]
@richardjgowers @jbarnoud @kain88-de Something has gone wrong while merging my PRs.

All the builds on the `develop` branch and my PR are failing with the following error -

```
import MDAnalysis as mda
E     File "/Users/utkbansal/Code/mdanalysis/package/MDAnalysis/__init__.py", line 148
E       import logging
E            ^
E   SyntaxError: invalid syntax
```

**Comment by jbarnoud** [comment]
@kain88-de @richardjgowers This is the commit responsible for the error. [Line 146 of `__init__.py`](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/__init__.py#L146) is broken.

**Comment by jbarnoud** [comment]
I forgot to paste the link to the commit: https://github.com/MDAnalysis/mdanalysis/commit/d99abba565f051747bbd8dd0b1139e74cd91b02e

**Comment by richardjgowers** [comment]
We won't revert 1414

**Comment by utkbansal** [comment]
This should be ready too once the build completes.

File: .travis.yml (hunk 3 above)
Comment (richardjgowers) [inline .travis.yml]: So here do we need to list `pytest-cov`?  And don't we need to list `pytest`?

**Comment by richardjgowers** [comment]
@utkbansal needs a rebase thanks to utils

**Comment by utkbansal** [comment]
Now how did utils slip away!

**Comment by utkbansal** [comment]
@richardjgowers Got it back.

**Comment by utkbansal** [comment]
@richardjgowers Looks good to go. The full build fails due to the time limit though.

**Pull request merged**